core/src/main/groovy/com/novoda/gradle/release/AndroidArtifacts.groovy

```diff
@@ -1,42 +1,51 @@
 package com.novoda.gradle.release
 
 import org.gradle.api.Project
+import org.gradle.api.Task
+import org.gradle.api.component.SoftwareComponent
 import org.gradle.api.tasks.bundling.Jar
 import org.gradle.api.tasks.javadoc.Javadoc
 
 class AndroidArtifacts implements Artifacts {
 
     def variant
 
+    // TODO: Declare that variant is
+    // https://google.github.io/android-gradle-dsl/current/com.android.build.gradle.LibraryExtension.html#com.android.build.gradle.LibraryExtension:libraryVariants
     AndroidArtifacts(variant) {
         this.variant = variant
     }
 
+    @Override
     def all(String publicationName, Project project) {
         [sourcesJar(project), javadocJar(project), mainJar(project)]
     }
 
     def sourcesJar(Project project) {
-        project.task(variant.name + 'AndroidSourcesJar', type: Jar) {
-            classifier = 'sources'
+        String taskName = variant.name + 'AndroidSourcesJar'
+        project.task(taskName, type: Jar) {
+            (it as Jar).classifier = 'sources'
             variant.sourceSets.each {
                 from it.java.srcDirs
             }
         }
     }
 
     def javadocJar(Project project) {
-        def androidJavadocs = project.task(variant.name + 'AndroidJavadocs', type: Javadoc) {
+        String taskName = variant.name + 'AndroidJavadocs'
+        Task androidJavadocs = project.task(taskName, type: Javadoc) {
             variant.sourceSets.each {
                 delegate.source it.java.srcDirs
             }
-            classpath += project.files(project.android.getBootClasspath().join(File.pathSeparator))
-            classpath += variant.javaCompile.classpath
-            classpath += variant.javaCompile.outputs.files
+            def javadoc = it as Javadoc
+            javadoc.classpath += project.files(project.android.getBootClasspath().join(File.pathSeparator))
+            javadoc.classpath += variant.javaCompile.classpath
+            javadoc.classpath += variant.javaCompile.outputs.files
         }
 
-        project.task(variant.name + 'AndroidJavadocsJar', type: Jar, dependsOn: androidJavadocs) {
-            classifier = 'javadoc'
+        String taskNameJar = variant.name + 'AndroidJavadocsJar'
```

Comment:
Do we really need to extract these `taskName`s? If it obvious that these are `Strings`, isn'it?

Reply:
I'm not sure if they are exactly the same. Here, since we are concatenating with a `String`, IDE can even infer the type. For example, if you do `def taskNameJar` here and tap `F1` on the variable, you will see the inferred type is already a `String`. 

Instead of this, I think I would do the following. It would be less verbose, no local variables.

```
project.tasks.create("${variant.name}AndroidJavadocsJar", Jar) {
```